lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -5055,6 +5044,13 @@ struct ExportVerilogPass : public ExportVerilogBase<ExportVerilogPass> {
     // TODO: This should be moved up to circt-opt and circt-translate.
     applyLoweringCLOptions(getOperation());
 
+    // Prepare the ops in the module for emission.
+    LoweringOptions options(getOperation());
+    mlir::OpPassManager preparePM("builtin.module");
+    auto &modulePM = preparePM.nest<hw::HWModuleOp>();
+    modulePM.addPass(createPrepareForEmissionPass(options));
+    (void)runPipeline(preparePM, getOperation());
```

Comment:
Is there a reason not to check if the pipeline had an error (similarly below)?
How about `if (failed(runPipeline(...))) return signalPassFailure();` or so?

Reply:
yea, added it.  It wasn't a failure possibility before (though probably should have been), so I wasn't worried about it.